tests/foreman/maintain/test_advanced.py

```diff
@@ -0,0 +1,434 @@
+"""Test class for satellite-maintain advanced command functionality
+
+:Requirement: foreman-maintain
+
+:CaseAutomation: Automated
+
+:CaseLevel: Component
+
+:CaseComponent: ForemanMaintain
+
+:Assignee: gtalreja
+
+:TestType: Functional
+
+:CaseImportance: Critical
+
+:Upstream: No
+"""
+import pytest
+import yaml
+
+from robottelo.config import robottelo_tmp_dir
+from robottelo.config import settings
+from robottelo.constants import MAINTAIN_HAMMER_YML
+
+pytestmark = pytest.mark.destructive
+
+
+# Common repositories for Satellite and Capsule
+common_repos = ['rhel-8-for-x86_64-baseos-rpms', 'rhel-8-for-x86_64-appstream-rpms']
+
+# Satellite repositories
+sat_611_repos = [
+    'satellite-6.11-for-rhel-8-x86_64-rpms',
+    'satellite-maintenance-6.11-for-rhel-8-x86_64-rpms',
+] + common_repos
+
+sat_612_repos = [
+    'satellite-6.12-for-rhel-8-x86_64-rpms',
+    'satellite-maintenance-6.12-for-rhel-8-x86_64-rpms',
+] + common_repos
+
+sat_613_repos = [
+    'satellite-6.13-for-rhel-8-x86_64-rpms',
+    'satellite-maintenance-6.13-for-rhel-8-x86_64-rpms',
+] + common_repos
+
+# Capsule repositories
+cap_611_repos = [
+    'satellite-capsule-6.11-for-rhel-8-x86_64-rpms',
+    'satellite-maintenance-6.11-for-rhel-8-x86_64-rpms',
+] + common_repos
+
+cap_612_repos = [
+    'satellite-capsule-6.12-for-rhel-8-x86_64-rpms',
+    'satellite-maintenance-6.12-for-rhel-8-x86_64-rpms',
+] + common_repos
+
+cap_613_repos = [
+    'satellite-capsule-6.13-for-rhel-8-x86_64-rpms',
+    'satellite-maintenance-6.13-for-rhel-8-x86_64-rpms',
+] + common_repos
+
+sat_repos = {
+    '6.11': sat_611_repos,
+    '6.12': sat_612_repos,
+    '6.13': sat_613_repos,
+}
+cap_repos = {
+    '6.11': cap_611_repos,
+    '6.12': cap_612_repos,
+    '6.13': cap_613_repos,
+}
+
+
+def test_positive_satellite_maintain_service_restart(sat_maintain):
+    """Restart service using advanced procedure run
+
+    :id: 64d3c78e-d602-43d6-bf77-f31135ed019e
+
+    :steps:
+        1. Run satellite-maintain advanced procedure run service-restart
+
+    :expectedresults: Satellite service should restart.
+    """
+    result = sat_maintain.cli.Advanced.run_service_restart()
+    assert 'FAIL' not in result.stdout
```

Comment:
Dont we have some `restarting` kind of keyword in stdout.

Also Can you somehow ensure the services are back to its state etc. Or just an hammer ping status ?

Reply:
we get FAIL in stdout whenever any service fails to start, and non-zero RC, so I think this checks are enough to test this command